### ISSUE TITLE
Backtesting results to include strategy's version

### ISSUE THREAD

**Issue by StephaneTurquay** [freqtrade/freqtrade#9727]
## Describe your environment

  * Freqtrade Version: 2023.12

## Describe the enhancement

**TLDR:** The enhancement would be to add the strategy version number, defined by `def version(self)`, in the backtesting results' JSON files so we can better compare the performances of various versions of our strategies.

**Full context:**
Hi there,

I'm developing a strategy, which I then run through hyperopt and finally goes through several rounds of backtesting on different time ranges.

Then I use the backtesting results (.json), from the `backtest_results` folder, which I add into a SQL database so I can best query the data and have better visualization with third-party tools.

The challenge that I face is that to compare the different backtesting results corresponding to different strategy versions, I would need to have its version number defined by `def version(self)`. 

You might have clever ideas to get to this objective, happy to hear your suggestions. Thank you 🙏

**Comment by stash86** [comment]
Why not just give them different class name?

**Comment by StephaneTurquay** [comment]
> Why not just give them different class name?

Correct me if I'm wrong, using different class name would require to change it in several places:

- In the strategy itself (class name)
- In my command line (now in my docker-compose.yml)

Ideally, I'd like to automate the hyperopt, backtesting, and insertion in the database therefore I might require quite some work to get there.

Do you have a better idea maybe to achieve that?

**Comment by stash86** [comment]
Well, if you want to automate hyperopt and change the version number as well, just add some lines to store the backtest result in a new file with the strategy version appended to the file name.

**Comment by StephaneTurquay** [comment]
Thanks @stash86! I'll try something as you suggested :-)
